pkg/controller/controller.go

```diff
@@ -620,6 +620,19 @@ func (c *controller) updateInstanceFinalizers(
 	instance *v1alpha1.Instance,
 	finalizers []string) error {
 
+	// Get the latest version of the instance so that we can avoid conflicts
+	// (since we have probably just updated the status of the instance and are
+	// now removing the last finalizer).
+	//
+	// It's important to do everything we can to ensure that we complete this
```

Comment:
Either remove the comment as is, because it's not at all clear what it means :) Or, alternatively, add the description there as you typed it up above. The way it reads now is not clear :)